UKHomeOffice/nmsw-ui
# NMSW-1313 Use peopleCount rather than FAL files to determine if declaration has people

# Ticket



----

## AC

We currently rely on checking if a FAL5 and of FAL6 file exists to determine if there are people on a declaration.

We now have a better way to determine people - using the response from /declaration we get a variety of ways to see the number of people & their status.

_AC_

Use people counts rather than FAL files to determine if declaration has people on it.
Use /declaration endpoint rather than /attachment endpoint

----

## To test

General Declaration page should redirect to task details when people exist on declaration

- create a new declaration
- upload a FAL1
- > task details should have a general declaration link
- click the link
- > you should be taken to the FAL 1 upload page
- upload a new FAL1
- upload a FAL5
- > task details should NOT have a general declaration link
- try to go to the general declaration url in your browser
- > you should be redirected back to the task details page

- create a new declaration
- upload a FAL1
- > task details should have a general declaration link
- click the link
- > you should be taken to the FAL 1 upload page
- upload a new FAL1
- upload a FAL6
- > task details should NOT have a general declaration link
- try to go to the general declaration url in your browser
- > you should be redirected back to the task details page
-  upload a FAL5
- > task details should NOT have a general declaration link
- try to go to the general declaration url in your browser
- > you should be redirected back to the task details page

----

## Notes
